nixos/modules/services/web-apps/openvscode-server.nix

```
@@ -0,0 +1,218 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.openvscode-server;
+  defaultUser = "openvscode-server";
+  defaultGroup = defaultUser;
+in {
+  options = {
+    services.openvscode-server = {
+      enable = mkEnableOption (lib.mdDoc "openvscode-server");
+
+      package = mkOption {
+        default = pkgs.openvscode-server;
+        defaultText = lib.literalExpression "pkgs.openvscode-server";
+        description = lib.mdDoc ''
+          Which openvscode-server derivation to use.
+        '';
+        type = types.package;
+      };
+
+      extraPackages = mkOption {
+        default = [ ];
+        description = lib.mdDoc ''
+          Packages that are available in the PATH of openvscode-server.
+        '';
+        example = "[ pkgs.go ]";
+        type = types.listOf types.package;
+      };
+
+      extraEnvironment = mkOption {
+        type = types.attrsOf types.str;
+        description = lib.mdDoc ''
+          Additional environment variables to passed to openvscode-server.
+        '';
+        default = { };
+        example = { PKG_CONFIG_PATH = "/run/current-system/sw/lib/pkgconfig"; };
+      };
+
+      extraArguments = mkOption {
+        default = [ ];
+        description = lib.mdDoc ''
+          Additional arguments that passed to openvscode-server
+        '';
+        example = ''[ "--log=info" ]'';
+        type = types.listOf types.str;
+      };
+
+      host = mkOption {
+        default = "127.0.0.1";
```

```suggestion
        default = "localhost";
```

I think we should use the same default host as upstream.